app/models/host_status/configuration_status.rb

```diff
@@ -78,7 +78,7 @@ def to_status(options = {})
     end
 
     def relevant?
-      host.configuration?
+      host.configuration? || last_report.present?
```

Comment:
That makes sense to me, if I explicitly enable always_show_configuration_status, I guess that warning is what I want to see if there's no report. The warning basically says that we are waiting for report to be uploaded.

I don't want to complicate this more so take it just as suggestion if users complain. Maybe we could take Puppet interval setting, add it to `created_at` time of host (if it's build == false) and display warning only if it's past this window. This wouldn't solve it for rebuilds though, we'd need built_at attribute.